templates/inspec/examples/google_appengine_standard_app_version/google_appengine_standard_app_version.erb

```diff
@@ -0,0 +1,14 @@
+<% gcp_project_id = "#{external_attribute('gcp_project_id', doc_generation)}" -%>
+<% gcp_location = "#{external_attribute('gcp_location', doc_generation)}" -%>
+<% standardappversion = grab_attributes['standardappversion'] -%>
+
+describe google_appengine_standard_app_version(project: <%= gcp_project_id -%>, location: <%= gcp_location -%>, version_id: <%= doc_generation ? "'#{standardappversion['version_id']}'" : "standardappversion['version_id']" -%>, service: <%= doc_generation ? "'#{standardappversion['service']}'" : "standardappversion['service']" -%>) do
+  it { should exist }
+  its('service') { should eq <%= doc_generation ? "'#{standardappversion['service']}'" : "standardappversion['service']" -%> }
+  its('name') { should eq <%= doc_generation ? "'#{standardappversion['name']}'" : "standardappversion['name']" -%> }
+  its('id') { should eq <%= doc_generation ? "'#{standardappversion['version_id']}'" : "standardappversion['version_id']" -%> }
```

Comment:
ya, its confusing. "version_id" is just used in self link. I have done override in inspec.yaml file. Also I have tested all these locally.

![image](https://user-images.githubusercontent.com/17652733/64058372-3b205700-cb78-11e9-8956-2299cde70355.png)

Reply:
We should remove any of these fields that `eq nil`. If `env_variables` is nil, that means that either `env_variables` is undefined on the resource or is not being returned by the API. We have specified `env_variables` via terraform, so that means that the field is not being returned by the API. If a field does not get returned by the API we shouldn't include it here or in the resource generation.

I would add 
```
  StandardAppVersion: !ruby/object:Overrides::Inspec::ResourceOverride
    properties:
      env_variables: !ruby/object:Overrides::Inspec::PropertyOverride
        exclude: true
      libraries: !ruby/object:Overrides::Inspec::PropertyOverride
        exclude: true
      runtime_api_version: !ruby/object:Overrides::Inspec::PropertyOverride
        exclude: true
      service: !ruby/object:Overrides::Inspec::PropertyOverride
        exclude: true
```

For version, it looks like in terraform the field is overridden via `terraform.yaml` to write id as `version_id`. Although I'm not sure why this is happening, I'd suggest either changing the `id` field defined in api.yaml to be `versionId` instead.

Reply:
I have not kept these variables in erb file. Only "service", "version_id" and "runtime" are getting validated. We should not exclude these other variables as many time it will come correct if it is set correctly in terraform. 

Forgot to push code to property override. I have adde those changes in inspec.yaml.
Can we work this to close initial pull. In case of any issue, we can take care in future.